### PR TITLE
FINERACT-646 Pockets API

### PR DESCRIPTION
**Pull request by ShruthiRajaram**

### CODE REVIEW

**Comment by vorburger** [comment]
@ShruthiRajaram I noticed you seem to be an active reviewer, do you know if this can be merged?

**Comment by santoshconflux** [comment]
@avikganguly01 and others , Could you please review it?

**Comment by ShruthiRajaram** [comment]
@vorburger , I worked on this feature last year. Since it is new feature I am waiting for feedback on this.  If anyone reviews  and approves it I will merger this PR.

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/self/pockets/service/AccountEntityServiceForShareAccountsImpl.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+	private final PlatformSecurityContext context;
+
+	@Autowired
+	public AccountEntityServiceForShareAccountsImpl(PlatformSecurityContext context) {
+		this.context = context;
+	}
+
+	@Override
+	public String getKey() {
+		// TODO Auto-generated method stub
```
Comment (vorburger) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/self/pockets/service/AccountEntityServiceForShareAccountsImpl.java]: This class looks un-finished... not sure it's a good idea to merge this with empty TODO method bodies? If perhaps you would consider removing this, then I think the rest could probably be merged (I would not mind to merge it).

**Review by vorburger — CHANGES_REQUESTED** [review]

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/self/pockets/service/AccountEntityServiceForShareAccountsImpl.java (hunk 1 above)
Comment (ShruthiRajaram) [reply fineract-provider/src/main/java/org/apache/fineract/portfolio/self/pockets/service/AccountEntityServiceForShareAccountsImpl.java]: @vorburger  , will update it soon Thanks for reviewing

**Comment by ShruthiRajaram** [comment]
@vorburger , I have completed the functionality for Shares now. Kindly review.

**Comment by vorburger** [comment]
@ShruthiRajaram now that we have Travis CI for all PRs, would you mind to `git rebase develop` this change so that we can see that it doesn't break the build?

**Pull request merged**